crypto/ex_data.c

```diff
@@ -522,8 +522,9 @@ static void int_free_ex_data(int class_index, void *obj, CRYPTO_EX_DATA *ad)
     for (i = 0; i < mx; i++) {
         if (storage[i] && storage[i]->free_func) {
             ptr = CRYPTO_get_ex_data(ad, i);
-            storage[i]->free_func(obj, ptr, ad, i,
-                                  storage[i]->argl, storage[i]->argp);
+	    if (ptr)
+            	storage[i]->free_func(obj, ptr, ad, i,
+                                      storage[i]->argl, storage[i]->argp);
```

Comment:
Unfortunately I cannot tell what function it is. All I can see is that whatever function it is, it breaks that tradition. So perhaps we can prevent misbehaving function from crashing the entire application by inserting this extremely simple check here?